src/type.ts

```diff
@@ -78,38 +77,30 @@ export class Type {
 			this.extend(options);
 	}
 
-	get identifier() {
+	get identifier(): Property {
 		if (this._identifier)
 			return this._identifier;
 		return this.baseType ? this.baseType.identifier : null;
 	}
 
-	set identifier(val) {
+	set identifier(val: Property) {
 		this._identifier = val;
 	}
 
```

Comment:
Both `createSync` and `create` will throw an error if asked to "create" an entity which already exists. This means the type for which you want to create a new instance already has an object pooled under the identifier specified in the provided `state`.

Reply:
Makes sense to me